tools/cpp/runfiles/runfiles_src.h

```diff
@@ -160,21 +180,48 @@ class Runfiles {
     return envvars_;
   }
 
+  // Returns a new Runfiles instance that by default uses the provided source
+  // repository as a default for all calls to Rlocation.
+  //
+  // The current instance remains valid.
+  Runfiles* WithSourceRepository(const std::string& source_repository) const {
```

Comment:
is it too late to make this return a `std::unique_ptr<Runfiles>`?

Reply:
The function is new and even though this deviates from `Create`, I agree it's just better.